### PR TITLE
ci: Run tests in CI

### PR DESCRIPTION
**Pull request by christopher-davis-afs**
This PE changes our CI configuration to run our tests when we receive a PR.
To keep consistency between libevse-security and other EVerest projects, we also
handle linting within `build_and_test.yaml`. Included is a commit to tweak the `.gitignore` file so that ci changes are not ignored.

One thing I noticed is that other components use ninja instead of make. Should we be using ninja here as well?

### CODE REVIEW

**Comment by christopher-davis-afs** [comment]
CC: @shankari

**Comment by christopher-davis-afs** [comment]
CC also @AssemblyJohn

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          extensions: hpp,cpp
+          exclude: cache
+      - name: Setup run scripts
+        run: |
+          mkdir scripts
+          rsync -a source/.ci/build-kit/ scripts
+      - name: Pull docker container
+        run: |
+          docker pull --quiet ghcr.io/everest/build-kit-alpine:latest
+          docker image tag ghcr.io/everest/build-kit-alpine:latest build-kit
```
Comment (shankari) [inline .github/workflows/build_and_test.yaml:39]: shouldn't this have `--platform` to be consistent with
https://github.com/EVerest/everest-demo/pull/8

Comment (christopher-davis-afs) [reply .github/workflows/build_and_test.yaml:39]: Done 👍

**Review by shankari — APPROVED** [review]

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        description: Which runner to use
+        type: choice
+        default: 'ubuntu-22.04'
+        required: true
+        options:
+          - 'ubuntu-22.04'
+          - 'large-ubuntu-22.04-xxl'
+  schedule:
+    - cron: '37 13,1 * * *'
+
```
Comment (couryrr-afs) [inline .github/workflows/build_and_test.yaml:16]: Just curious if it make sense to have this schedule in every lib?

**Comment by christopher-davis-afs** [comment]
@shankari I've made some changes to this based on the community feedback on the libocpp PR. Could you take another look at this?

CC: @Dominik-K as the provider for the aforementioned community review.

**Comment by andistorm** [comment]
When the pull request is ready for merging, please mark me as a reviewer.

**Review by andistorm — CHANGES_REQUESTED** [review]
Looks good to me in general, with a few minor comments.

Could you provide a link to a successful run of this workflow in your fork?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # libevse-security
 
+![Github Actions](https://github.com/EVerest/libevse-security/actions/workflows/build_and_test.yml/badge.svg)
+
```
[the remark this answers is not in the document]
Comment (christopher-davis-afs) [reply README.md:4]: Yes, I think so. This should work just fine once this is merged.

**Comment by christopher-davis-afs** [comment]
> Could you provide a link to a successful run of this workflow in your fork?

https://github.com/US-JOET/libevse-security/actions/runs/7264602730

File: .ci/build-kit/install_and_test.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+cmake \
+    -B build \
+    -S "$EXT_MOUNT/source" \
+    -G Ninja \
+    -DBUILD_TESTING_EVSE_SECURITY=ON \
```
Comment (andistorm) [inline .ci/build-kit/install_and_test.sh:9]:
```suggestion
    -DBUILD_TESTING=ON \
```

I thought running a workflow added by a fork is not possible. I don't understand why it is now.

This change should probably fix
https://github.com/EVerest/libevse-security/actions/runs/7817099985/job/21329584371?pr=25

Locally the unit tests pass..

**Review by andistorm — APPROVED** [review]

**Pull request merged**